### ISSUE TITLE
excludeIPRanges not working

### ISSUE THREAD

**Issue by michaelst** [istio/istio#30280]
**Bug description**
I am trying to have internal traffic not go through the proxy, I added this `--set values.global.proxy.excludeIPRanges="10.0.0.0/8"`, however, with or without that rule only local traffic is going through the proxy according to the logs.

[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
Outbound traffic to any an IP in the range specified should not go through the proxy.

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
```
client version: 1.8.1
control plane version: 1.8.1
data plane version: none
```

```
Client Version: v1.17.14-dispatcher
Server Version: v1.17.14-gke.400
```

**How was Istio installed?**
```
istioctl install --set profile=minimal \
 --set values.cni.cniBinDir=/home/kubernetes/bin \
 --set values.global.proxy.excludeIPRanges="10.0.0.0/8" \
 --set components.cni.enabled=true \
 --set components.cni.namespace=kube-system \
 --set meshConfig.outboundTrafficPolicy.mode=REGISTRY_ONLY \
 --set meshConfig.accessLogFile=/dev/stdout \
 --set meshConfig.accessLogEncoding=JSON
```

**Environment where the bug was observed (cloud vendor, OS, etc)**
GKE - 1.17.14-gke.400

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by jcmcken** [comment]
I think we may be hitting this issue too. In our case, it's ``includeIPRanges``. It seems like Istio CNI does not respect any global settings configured via the Istio operator and instead uses its own, unconfigurable defaults, leaving annotations as the only way to customize the settings when using the CNI method. Is there no way to configure cluster-wide defaults with Istio CNI?

The injector method works fine with the global proxy settings.

**Comment by bianpengyuan** [comment]
I can think of several approaches:
1. Add proxy config exclude/include ip range/port annotations to the pod at injection time, if they are not already defined. This is the easiest option and requires least changes, but can be confusing to user.
2. Pass in proxy config exclude/include ip range / port as CNI daemonset env variables during install time, and pass them to CNI configuration file. This requires CNI daemonset restarting for the updates of those field, but cleaner than 1. 
3. Mount mesh configuration, then pass it to CNI configuration file. For long run this will be the most ideal solution, as it will open more ability at CNI. Although I am not sure how to do this at GKE where CNI is installed at kube-system namespace, other than duplicating the config map.

@therealmitchconnors @stewartbutler @howardjohn wdyt?

**Comment by howardjohn** [comment]
I thought how this already worked was injecting an annotation on the pod that cni reads. Is that not happening? I think it was broken for a bit so may be fixed now but broken in the original report version (1.8)

**Comment by therealmitchconnors** [comment]
I'm in favor of adding this as a feature of CNI.  It seems like it may be a blocker for beta as well...

**Comment by howardjohn** [comment]
isn't it handled by https://github.com/istio/istio/blob/04156e94a9ee9f57e8394d9b4fba65af1889b41d/manifests/charts/istio-control/istio-discovery/files/injection-template.yaml#L20?

**Comment by bianpengyuan** [comment]
Yeah, just verified this is handled at 1.9, with injector v2 work after we pass through port/ip annotations.

@michaelst @jcmcken Could you please upgrade and try Istio 1.9?

**Comment by howardjohn** [comment]
It might have been fixed in 1.8.x as well. I vaguely recall some PRs. I can
try to dig them up

**Comment by bianpengyuan** [comment]
Found it: https://github.com/istio/istio/pull/28864 Although seems like not in 1.8.

**Comment by bianpengyuan** [comment]
And looks like pretty hard to backport..

**Comment by bianpengyuan** [comment]
Close as this is fixed at 1.9.
